### DISCUSSION TITLE
<Script> tag does not produce <script> tag when placed in root layout

### DISCUSSION THREAD

**Discussion by figuernd** [vercel/next.js#56626] [Help]
### Summary

Following the documentation here: [https://nextjs.org/docs/pages/api-reference/components/script#beforeinteractive](https://nextjs.org/docs/pages/api-reference/components/script#beforeinteractive)

The example shown in `app/layout.tsx` differs considerably from the boilerplate given by create-next-app.

From the docs above:
```
import { Html, Head, Main, NextScript } from 'next/document'
import Script from 'next/script'
 
export default function Document() {
  return (
    <Html>
      <Head />
      <body>
        <Main />
        <NextScript />
        <Script
          src="https://example.com/script.js"
          strategy="beforeInteractive"
        />
      </body>
    </Html>
  )
}
```

What you're given by create-next-app:
```
import './globals.css'
import type { Metadata } from 'next'
import { Inter } from 'next/font/google'

const inter = Inter({ subsets: ['latin'] })

export const metadata: Metadata = {
  title: 'Create Next App',
  description: 'Generated by create next app',
}

export default function RootLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <html lang="en">
      <body className={inter.className}>{children}</body>
    </html>
  )
}
```

I tried to adapt this by inserting <Script> tags into the body like so:
```
      <body className={inter.className}>
        {children}
        <Script src="js/nexus.js" strategy="beforeInteractive" />
      </body>
```

But no corresponding script appears in the head of the document, as the documentation suggests should happen.

**Comment by Hillsie** [comment]
I can also confirm that the `strategy="beforeInteractive"` doesn't work for me. 
The linter says that it should be in the `pages/_document.ts|js`. I tried adding it there and this didn't work.



# Example `beforeInteractive`
```javascript
<html lang="en">
    <Script  id="console-log"  strategy="beforeInteractive">
        {`(() => {
          console.log('hello beforeInteractive script tag');
        })();`}
    </Script>
    <body>
        <main>
          {children}
        </main>
    </body>
</html>
```
Linter message : https://nextjs.org/docs/messages/no-before-interactive-script-outside-document

<details>
    <summary><h1>Console error log</h1></summary>
    <pre>
    Warning: Cannot render a sync or defer <script> outside the main document without knowing its order. Try adding async="" or moving it into the root <head> tag.
Script@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/script.js:175:130
html
RedirectErrorBoundary@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/redirect-boundary.js:72:9
RedirectBoundary@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/redirect-boundary.js:80:24
NotFoundErrorBoundary@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/not-found-boundary.js:54:9
NotFoundBoundary@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/not-found-boundary.js:62:62
DevRootNotFoundBoundary@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/dev-root-not-found-boundary.js:32:24
ReactDevOverlay@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/react-dev-overlay/internal/ReactDevOverlay.js:66:9
HotReload@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/react-dev-overlay/hot-reloader-client.js:294:37
Router@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/app-router.js:157:93
ErrorBoundaryHandler@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/error-boundary.js:82:9
ErrorBoundary@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/error-boundary.js:110:53
AppRouter@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/app-router.js:440:47
ServerRoot@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/app-index.js:126:24
RSCComponent
Root@webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/app-index.js:142:24
    </pre>
</details>


However, `strategy="afterInteractive"` works fine. 


# Example `afterInteractive`

```javascript
<html lang="en">
    <Script  id="console-log"  strategy="afterInteractive">
        {`(() => {
          console.log('hello afterInteractive script tag');
        })();`}
    </Script>
    <body>
        <main>
          {children}
        </main>
    </body>
</html>
```
Would be really handy for a feature I am looking to build to have it `beforeInteractive`.

**Comment by Hillsie** [comment]
Looks like there's a storm of issues around this feature.
https://github.com/vercel/next.js/issues?q=is%3Aissue+is%3Aopen+beforeInteractive

might be worth closing this off and watching checking in future versions.
